client/state/happychat/middleware.js

```diff
@@ -111,47 +103,32 @@ export const connectChat = ( connection, { getState, dispatch } ) => {
 		return;
 	}
 
+	const url = config( 'happychat_url' );
+
 	const user = getCurrentUser( state );
 	const locale = getCurrentUserLocale( state );
 	let groups = getGroups( state );
-
-	// update the chat locale and groups when happychat is initialized
 	const selectedSite = getHelpSelectedSite( state );
 	if ( selectedSite && selectedSite.ID ) {
 		groups = getGroups( state, selectedSite.ID );
 	}
 
-	// Notify that a new connection is being established
-	dispatch( setConnecting() );
-
-	debug( 'opening with chat locale', locale );
-
-	// Before establishing a connection, set up connection handlers
-	connection
-		.on( 'connected', () => {
-			dispatch( setConnected() );
-
-			// TODO: There's no need to dispatch a separate action to request a transcript.
-			// The HAPPYCHAT_CONNECTED action should have its own middleware handler that does this.
-			dispatch( requestChatTranscript() );
-		} )
-		.on( 'disconnect', reason => dispatch( setDisconnected( reason ) ) )
-		.on( 'reconnecting', () => dispatch( setReconnecting() ) )
-		.on( 'message', event => dispatch( receiveChatEvent( event ) ) )
-		.on( 'status', status => dispatch( setHappychatChatStatus( status ) ) )
-		.on( 'accept', accept => dispatch( setHappychatAvailable( accept ) ) );
-
-	// create new session id and get signed identity data for authenticating
-	return startSession()
-		.then( ( { session_id, geo_location } ) => {
-			if ( geo_location && geo_location.country_long && geo_location.city ) {
-				dispatch( setGeoLocation( geo_location ) );
-			}
+	const happychatUser = {
+		signer_user_id: user.ID,
+		locale,
+		groups,
+	};
+
+	const auth = () =>
+		startSession()
+			.then( ( { session_id, geo_location } ) => {
+				happychatUser.geo_location = geo_location;
+				return sign( { user, session_id } );
+			} )
+			.then( ( { jwt } ) => Promise.resolve( Object.assign( { jwt }, happychatUser ) ) )
```

Comment:
I guess we could just use `{ jwt, ...happychatUser }` here.

Reply:
On second thought, even though separating this by using a separate `auth` method makes sense, maybe we could merge `init` and `open` methods and just call it from here like it was before. The advantage would be that we eliminate the double promise chain (from auth and init) which can become difficult to follow. Just a suggestion non blocking.

Reply:
My thinking was whether the whole auth should be owned by the Connection (including the REST calls part) as that is also part of the auth flow. That should be given a bit more thought, so, at the moment, I changed it to receive user data instead.